parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -409,6 +409,13 @@ write_files:
   content: !!binary |
     {{GetVariableProperty "cloudInitData" "ensureIMDSRestrictionScript"}}
 
+- path: /opt/azure/containers/validate-kubelet-credentials.sh
+  permissions: "0755"
+  encoding: gzip
+  owner: root
+  content: !!binary |
```

Comment:
what does this do ? we pass the script from the CustomData ?

Reply:
this adds the file within our customdata payload - until we're confident that newer CSE/customdata is never being executed on older VHDs, this is required as to not take a dependency on VHD version